executor/prepared.go

```diff
@@ -276,6 +286,34 @@ func (e *ExecuteExec) Build() error {
 	return nil
 }
 
+// getPhysicalPlan allows us to reuse the compiled plan of a prepared statement in the same session.
+func (e *ExecuteExec) getPhysicalPlan(ctx context.Context, is infoschema.InfoSchema, pstmtID uint32, schemaVer int64, node ast.Node, useCache bool) (plan.Plan, error) {
+	var key string
+
+	ctx.GetSessionVars().StmtCtx.UseCache = useCache
+	if useCache {
+		key = key4PreparedStmt(ctx, pstmtID, schemaVer)
+		if cacheEntry, exists := ctx.PlanCache().Get(key); exists {
+			return cacheEntry.(plan.Plan), nil
+		}
+	}
+	p, err := plan.Optimize(ctx, node, is)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if useCache {
+		ctx.PlanCache().Set(key, p.(interface{}))
+	}
+
+	return p, err
+}
+
+// key4PreparedStmt converts prepared statement ID with its context to a formatted string
+func key4PreparedStmt(ctx context.Context, pstmtID uint32, schemaVer int64) string {
+	return ctx.GetSessionVars().CurrentDB + ":" + strconv.FormatUint(ctx.GetSessionVars().ConnectionID, 10) + ":" +
+		strconv.FormatUint(uint64(pstmtID), 10) + ":" + strconv.FormatInt(schemaVer, 10)
```

Comment:
Schema version is a global variable shared by all the TiDB sessions, any ddl operation in any session will change the schema version number. It means that after a ddl operation accomplished, plan cache in every session can be cleared, because they can never be hit in the future.

Reply:
I agree what you pointed out. In order to increase the cached plans' lifetime, the current schema version management scheme should allow us to know whether or not the schema only related to the cached plan is changed. We need to handle this issue as another feature.